mods/default/torch.lua

```diff
@@ -0,0 +1,132 @@
+
+--[[
+
+Torch mod - formerly mod "Torches"
+======================
+
+(c) Copyright BlockMen (2013-2015)
+(C) Copyright sofar <sofar@foo-projects.org> (2016)
+
+This mod changes the default torch drawtype from "torchlike" to "mesh",
+giving the torch a three dimensional appearance. The mesh contains the
+proper pixel mapping to make the animation appear as a particle above
+the torch, while in fact the animation is just the texture of the mesh.
+
+
+License:
+~~~~~~~~
+(c) Copyright BlockMen (2013-2015)
+
+Textures and Meshes/Models:
+CC-BY 3.0 BlockMen
+Note that the models were entirely done from scratch by sofar.
+
+Code:
+Licensed under the GNU LGPL version 2.1 or higher.
+You can redistribute it and/or modify it under
+the terms of the GNU Lesser General Public License
+as published by the Free Software Foundation;
+
+You should have received a copy of the GNU Lesser General Public
+License along with this library; if not, write to the Free Software
+Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+
+See LICENSE.txt and http://www.gnu.org/licenses/lgpl-2.1.txt
+
+--]]
+
+minetest.register_node("default:torch", {
+	description = "Torch",
+	drawtype = "mesh",
+	mesh = "torch_floor.obj",
+	inventory_image = "default_torch_on_floor.png",
+	wield_image = "default_torch_on_floor.png",
+	tiles = {{
+		    name = "default_torch_on_floor_animated.png",
+		    animation = {type = "vertical_frames", aspect_w = 16, aspect_h = 16, length = 3.3}
+	}},
+	paramtype = "light",
+	paramtype2 = "wallmounted",
+	sunlight_propagates = true,
+	walkable = false,
+	liquids_pointable = false,
+	light_source = 13,
+	groups = {choppy=2, dig_immediate=3, flammable=1, attached_node=1, torch=1},
+	drop = "default:torch",
+	selection_box = {
+		type = "wallmounted",
+		wall_top = {-1/16, -2/16, -1/16, 1/16, 0.5, 1/16},
+		wall_bottom = {-1/16, -0.5, -1/16, 1/16, 2/16, 1/16},
+	},
+	sounds = default.node_sound_wood_defaults(),
+	on_place = function(itemstack, placer, pointed_thing)
+		if pointed_thing.type ~= "node" then
+			-- no interaction possible with entities, for now.
+			return itemstack
+		end
+
+		local under = pointed_thing.under
+		local node = minetest.get_node(under)
+		local def = minetest.registered_nodes[node.name]
+		if def and def.on_rightclick then
```

Comment:
Don't we then want to pass on the `nil` through if that's the case? It's up to the itemframe to account for a `nil` placer (e.g. a "bot" places an item in an item frame).

`and ((not placer) or (placer and not placer:get_player_control().sneak))`